mono/mini/interp/interp.c

```diff
@@ -3967,7 +3970,8 @@ interp_exec_method_full (InterpFrame *frame, ThreadContext *context, FrameClause
 			stackval *retval;
 			gboolean is_void = *ip == MINT_VCALL || *ip == MINT_VCALLVIRT;
 			gboolean is_virtual = *ip == MINT_CALLVIRT || *ip == MINT_VCALLVIRT;
-			gpointer native_stack_addr = frame->native_stack_addr ? (gpointer)((guint8*)frame->native_stack_addr - 1) : (gpointer)&retval;
+			// FIXME This assumes a grow down stack.
+			native_stack_addr = frame->native_stack_addr ? frame->native_stack_addr - 1 : native_stack_addr;
```

Comment:
Does this conditionality make sense, or should frame->native_stack_addr always be set?